Fix TestNonprintingCharacters tests

shilpakodli opened the pull request:
<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Describe Bug or Feature
Few tests from TestNonprintingCharacters test suite were failing with below error. This is due to on few machines, default shell for pbsusers is cshell and thus it was throwing error while reading job's outputfile as a pbsuser.
AssertionError: 'Warning: no access to tty (Bad file descr[66 chars]31mY' != 'VAR_IN_TERM=X^[[1m^[[31mY'
   Warning: no access to tty (Bad file descriptor).
   Thus no job control in this shell.
   VAR_IN_TERM=X^[[1m^[[31mY

Tests failed due to above error:
- test_nonprint_character_default_qsubV
- test_nonprint_character_job_array
- test_nonprint_character_qsubV
- test_terminal_control_default_qsubV
- test_terminal_control_job_array
- test_terminal_control_qsubV

test_nonprint_character_in_qstat_json_valid test was failing with json load error:
AssertionError: JSONDecodeError('Invalid \\escape: line 38 column 49 (char 1391)',) is not false

#### Describe Your Change
For "no access to tty" error, added extra parameter while submitting job "ATTR_S: '/bin/bash'". This will run job in bash shell.
For json load error, removed escape character from passing environment variable list.



#### Attach Test and Valgrind Logs/Output
[TestNonprintingCharacters_afterfix.txt](https://github.com/openpbs/openpbs/files/5035940/TestNonprintingCharacters_afterfix.txt)
[TestNonPrintingCharacters_before_fix.txt](https://github.com/openpbs/openpbs/files/5035944/TestNonPrintingCharacters_before_fix.txt)


<!--- Pull Request Guidelines: [Pull Request Guidelines](https://pbspro.atlassian.net/wiki/spaces/DG/pages/1187348483/Pull+Request+Guidelines) -->

hirenvadalia commented on test/tests/functional/pbs_nonprint_characters.py:
```diff
@@ -606,13 +606,17 @@ def test_nonprint_character_in_qstat_json_valid(self):
                 self.logger.info('##### excluded char: %s' % repr(ch))
                 continue
-            a = {ATTR_v: r"var1=\'A\,B\,%s\,C\,D\'" % ch}
+            if uhost is None or self.du.is_localhost(uhost):
```
U don't need `uhost is None` check, as `is_localhost` will return True is uhost is None

shilpakodli commented:
Thanks Hiren for the review. I have addressed your comments. Please find latest test execution logs
[test_nonprint_character_in_qstat_json_valid_latest_logs.txt](https://github.com/openpbs/openpbs/files/5036103/test_nonprint_character_in_qstat_json_valid_latest_logs.txt)

bayucan commented:
Merged #1956 to master.
